crates/project/src/project.rs

```diff
@@ -11692,7 +11692,46 @@ fn include_text(server: &lsp::LanguageServer) -> bool {
         .unwrap_or(false)
 }
 
+async fn load_direnv_environment(dir: &Path) -> Result<Option<HashMap<String, String>>> {
+    let direnv_res = smol::process::Command::new("direnv")
```

Comment:
Hmmm, what I don't like about this approach is that we always try to run direnv, even though it might not be installed. It's not a big deal from a performance perspective or anything, but I think checking in `PATH` whether direnv exists first is the better approach.

Reply:
So `smol::process::Command::new` returns `ErrorKind::NotFound` _before_ it creates a process? Or does it create the process and returns the error from `exec`?

Reply:
That's a hard question and the answer is: I don't know... But I also kinda don't know about any other cross platform way to check if program is in PATH so I'd say that it's probably fine, as spawning processes (not shells) is pretty cheap overall

Reply:
There's this: https://crates.io/crates/which We already use that to check for language server binaries in the environment returned from this function here.

Reply:
Oh, we already have it as a dependency? I didn't know about that, so I've avoided using it, sorry

Reply:
Okay, I've added a check